addons/sale/wizard/sale_order_discount.py

```diff
@@ -24,6 +24,11 @@ class SaleOrderDiscount(models.TransientModel):
         ],
         default='sol_discount',
     )
+    tax_ids = fields.Many2many(
+        comodel_name='account.tax',
+        string="Tax",
+        context={'active_test': False},
```

Comment:
I don't think we want to show inactive taxes here ;).
You can also move the domain in the field definition here instead of in the view.

(same remarks for the taxes field on the loyalty reward ;))